### PR TITLE
add bleed trail and cosmic ray code

### PR DESCRIPTION
**Pull request by jchiang87**
This is the code from the old imSim.

### CODE REVIEW

**Comment by jchiang87** [comment]
@rmjarvis I tried to implement most of your suggestions, but my understanding of the galsim config stuff is pretty limited, so I expect these changes may not be the preferred way.

File: config/imsim.yaml
Comment (rmjarvis) [inline config/imsim.yaml:239]: I think this is a syntax error.  To leave it empty, I'm pretty sure you need it to be `{}`.

Comment (rmjarvis) [reply config/imsim.yaml:239]: Actually, this is apparently fine. I just tried it, and it worked. Huh. Yaml is cleverer than I thought... :)

Comment (cwwalter) [reply config/imsim.yaml:239]: Hmm.. that's surprising... I had to use the {} in the camera geometry case or it didn't work.

Comment (rmjarvis) [reply config/imsim.yaml:239]: I think with nothing there, YAML sets it to None, not an empty dict, which in this case is fine, but wouldn't be for camera_geometry.  In any case, I think it's probably cleaner to remove this line, since most users will be fine with the default CR file I suspect.

File: imsim/bleed_trails.py
Comment (rmjarvis) [reply imsim/bleed_trails.py:118]: I'll make a unit test that checks the thing I'm concerned about on my unit test branch and fix it there if the test confirms my thought about this.

File: config/imsim.yaml
Comment (cwwalter) [reply config/imsim.yaml:239]: I see.

So, if you used the {} would it overwrite the default so things wouldn't work?

In the old config we had 

```
catalog = default
```

which is at least good in letting people know the option is there.

Comment (rmjarvis) [reply config/imsim.yaml:239]: Right, but that nominally means to use the catalog file named `default`.  So IMO is more confusing than just leaving it out, which means to use the default value, which is some catalog we ship with imsim.

**Review by rmjarvis — APPROVED** [review]

**Comment by rmjarvis** [comment]
I made one change to how some of the config things are parsed, which I think gives better error messaging when the user does something wrong.
I also fixed the det_name split to split on '_', rather than '-'.  I think Chris noticed this on main, and it tripped when I tested my own changes for this branch.
And I rebased to master, so it should merge cleanly now.

I think this is all ready to go.

File: config/imsim.yaml
Comment (cwwalter) [reply config/imsim.yaml]: Right.. I actually meant just to leave it blank with a comment as a reminder it is settable.  But, anyway I agree that almost no one will ever touch it.

**Comment by jchiang87** [comment]
Thanks @rmjarvis!  I'll go ahead and merge ~COB PT today.

**Comment by cwwalter** [comment]
Great!

Jim, I just merged the split config files. Could you update the new imsim-config.yaml file with the new yaml for these changes too?

(At some point I want to  delete the old monolithic imsim.yaml file too)

**Comment by jchiang87** [comment]
> Jim, I just merged the split config files. Could you update the new imsim-config.yaml file with the new yaml for these changes too?

Would you mind doing it?  I haven't been following the details of your changes, so I think it would be easier if you did that. Thanks.

**Comment by cwwalter** [comment]
Sure. 

I think I can wait until you merge and then just copy your changes from that section of the monolithic file into my yaml file.

**Comment by jchiang87** [comment]
@cwwalter I rebased to main and made the change to imsim-config.yaml.   If everything looks ok to you, please go ahead and merge.

**Comment by cwwalter** [comment]
Thanks! Sorry I was out before.

So only change to the yaml files is the two cosmic ray lines right?

**Pull request merged**
